app/modules/recordings.ts

```diff
@@ -9,15 +9,30 @@ export const reducer = (state = INITIAL_POPUP_STATE, action: Action) => {
     case actionType.START_LISTENING:
       return {...state, enabled : true}
     case actionType.UPDATE_FIELD:
-      return {...state, ...action};
+      return {...state, [action.name]: action.value };
     case actionType.UPDATE_FIELDS:
       return {...state,  ...action.payload};
     case actionType.STOP_LISTENING:
       return {...state, enabled: false};
     case actionType.ERROR:
       return {...state, errorMessage: action.errorMessage }
-    case actionType.CLEAR_REQUESTS : return {...state, requests: [] }
+    case actionType.CLEAR_REQUESTS :
+      return {...state, requests: [] }
+    case actionType.TOGGLE_CHECKBOX:
+      let selectedReqIds = state.selectedReqIds;
+      if (action.checked) {
+        selectedReqIds = [ ...selectedReqIds, action.reqId ]
+        return {...state , selectedReqIds, checked : !action.checked }
+      } else {
+        removeReqId(state, action)
```

Comment:
As discussed don't pass the whole state and action. Pass only the minimum needed items.

Reply:
The newState is being returned from the function on L37 as https://github.com/code-mancers/interceptor/blob/0c253454b3d57d02aa2e7abfee18d50d2f1a573e/app/modules/recordings.ts#L37 and all the operations relating to state are done inside the function unlike earlier as 
```js
{...state, removeReqId(state, action) } 
```

Still recommend to do the earlier way as in 
```js
{...state, removeReqId(state.selectedReqIds, action.reqId) } 
``` 
?